tests/api/test_record_permissions.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test record permissions."""
+import uuid
+
+import pytest
+from flask_principal import RoleNeed, identity_loaded
+from flask_security import login_user
+from invenio_accounts.models import User
+from invenio_records.api import Record
+
+from invenio_app_ils.records.permissions import has_admin_permission, \
+    has_librarian_permission, record_permission_factory
+
+
+@pytest.mark.parametrize('access,action,is_allowed', [
+    ({'foo': 'bar'}, 'read', True),     # default cases
+    ({'foo': 'bar'}, 'update', False),
+    ({'_access': {'read': [1, 'no-access@invenio',
+                           'no-access-either@invenio']}}, 'read', True),
+    ({'_access': {'read': [2, 'no-access@invenio']}}, 'read', False),
+    ({'_access': {'read': ['test@invenio']}}, 'read', True),
+    # permission for specific user to create
+    ({'_access': {'update': [1]}}, 'update', True),
+    # checks if the access works for different actions
+    ({'_access': {'update': [1]}}, 'create', False),
+    ({'_access': {'delete': [1]}}, 'update', False),
+
+    # delete access for user and librarian
+    ({'_access': {'delete': [1, 'librarian']}}, 'delete', True),
+
+])
+def test_record_generic_access(db, users, access, action, is_allowed):
+    """Test access control for records."""
+
+    @identity_loaded.connect
+    def mock_identity_provides(sender, identity):
+        """Add additional role to the user."""
+        roles = [RoleNeed('test@invenio')]
+        identity.provides |= set(roles)
```

Comment:
😯 I would say it is less common to use the bitwise OR operator, compared to `+=` or `-=`. I would add a comment explaining what we are trying to do here.

Reply:
+= operator does not work for sets